# Data Science: Individual Planning Report
This R notebook will perform the individual planning stage of the data science project. The main pursuits of this report include understanding the data, formulating a specific inquiry derived from one of the three broad questions, creating several visualizations to foster greater comprehension of the data and foreseeable roadblocks, and selecting an appropriate method to address my question of interest.

In [1]:
### Run this cell before continuing to load in the necessary R packages.
library(tidyverse)
library(repr)
library(rvest)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors

Attaching package: ‘rvest’


The following object is masked from ‘package:readr’:

    guess_encoding




In [2]:
# Read in datasets directly from their raw URLs obtained from the GitHub repository

# Read in player.csv file
url_players <- "https://raw.githubusercontent.com/Adrian12-max/Data-Science-Project-Individual-Planning-Stage/refs/heads/main/players.csv"
players <- read_csv(url_players) |>
head()
players

# Read in sessions.csv file
url_sessions <- "https://raw.githubusercontent.com/Adrian12-max/Data-Science-Project-Individual-Planning-Stage/refs/heads/main/sessions.csv"
sessions <- read_csv(url_sessions) |>
head()
sessions

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

## 1. Data Description

**players.csv:** provides a list including demographic information and self-reported play statistics for each player. Each row corresponds to a unique player. This tibble contains 196 observations/rows and 7 variables/columns. The variables include:

- `experience` (character): Self-reported playing experience: *Newbie, Amateur, Regular, Veteran, or Pro*
- `subscribe` (logical): Whether the player is subscribed to the game-related newspaper (evaluates to either TRUE or FALSE)
- `hashedEmail` (character): A unique hashed identifier used to link sessions to players
- `played_hours` (double): Total number of hours the player has spent on the server
- `name` (character): The player's self-reported personal name
- `gender` (character): The player's self-reported gender 
- `Age` (double): The age of the player in years
  
**sessions.csv:** provides a list of all individual play sessions by each player, including data about the session. Each row records one play session for a player. This tibble contains 1535 observations/rows and 5 variables/columns. The variables include:

- `hashedEmail` (character): A unique hashed identifier used to link sessions to players
- `start_time` (character): The local date-time (in year-month-day hour:minute format) representing when the session began
- `end_time` (character): The local data-time (in year-month-day hour:minute format) representing when the session ended
- `original_start_time` (double): Numeric timestamp (in milliseconds since 1970-01-01 00:00:00 UTC) indicating when the session began
- `original_end_time` (double): Numeric timestamp (in milliseconds since 1970-01-01 00:00:00 UTC) indicating when the session ended

**Summary Statistics**

| Variable       | Mean  | Min  | Max    | Sum     |
| -------------- | ----- | ---- | ----   | ------  |
| `Age`          | 21.14 | 9.00 | 58.00  | 4101.00 |
| `played_hours` | 5.90  | 0.00 | 223.10 | 1145.80 | 


## 2. Questions

**Selected Broad Question:** We would like to know which "kinds" of players are most likely to contribute a large amount of data so that we can target those players in our recruiting efforts.

**Specific Inquiry:** Can `Age`, `gender`, and `experience` predict the total number of hours played (`played_hours`) by a player in the `players.csv` dataset?

**How will the data help address this question?**

The `players.csv` dataset contains demographic information for each player, including:
- `Age` (double)
- `gender` (character)
- `experience` (character)
- `played_hours` (double)

This inquiry uses `played_hours` as the response variable and `Age` (numeric), `gender` (character), and `experience` (character) as the predictor variables. By default, the `gender` and `experience` data are stored as character strings; however, they actually represent categories (`Male`, `Female`, `Non-Binary`, etc. for `gender` and `Newbie`, `Amateur`, `Regular`, etc. for `experience`). Before running the analysis, these character variables should be converted to a factor type to ensure that the selected regression model runs properly and interprets these variables as categorical rather than arbitrary strings. Conversely, the numeric predictor variable `Age` should be retained as is because it represents a measurable, continuous quantity that can be directly included in mathematical models such as regression.

## 3. Exploratory Data Analysis and Visualization

## 4. Methods and Plans

## 5. GitHub Repository